src/g_utils.c

```diff
@@ -2814,3 +2816,33 @@ char* make_dots(char *dots, size_t dots_len, int cmd_max_len, char *cmd)
 	dots[len] = 0;
 	return dots;
 }
+
+void trim_non_alphanumeric(char *input)
+{
+	char *src = input;
+	char *dst = input;
+
+	while (*src) {
+		if (isalnum(*src))
+			*dst++ = *src;
+		src++;
+        }
+
+	*dst = '\0';
+}
+
+int sort_alphanumeric(const void *a, const void *b)
+{
+	char *aa = strdup(a);
```

Comment:
You can’t use heap memory allocation in the mod, use stack, not to mention you doesn’t check success here.

Reply:
Aha, I wasn't aware that I couldn't use the heap. I'll fix that! (Sorry, it was very sloppy of me to not check if the allocation succeeded)